contracts/src/stdlib/IndexerModule.sol

```diff
@@ -0,0 +1,170 @@
+   */
+  function indexAttestation(bytes32 attestationId) public {
+    AttestationRegistry attestationRegistry = AttestationRegistry(router.getAttestationRegistry());
+    Attestation memory attestation = attestationRegistry.getAttestation(attestationId);
+    _indexAttestation(attestation);
+  }
+
+  /**
+   * @dev Indexes multiple attestations.
+   * @param attestationIds An array of attestation IDs to index.
+   */
+  function indexAttestations(bytes32[] calldata attestationIds) external {
+    uint256 length = attestationIds.length;
+    for (uint256 i = 0; i < length; i++) {
+      indexAttestation(attestationIds[i]);
+    }
+  }
+
+  /**
+   * @dev Returns the attestation IDs for a given subject.
+   * @param subject The subject to retrieve attestation IDs for.
+   * @return An array of attestation IDs.
+   */
+  function getAttestationIdsBySubject(bytes memory subject) public view returns (bytes32[] memory) {
+    return attestationIdsBySubject[subject];
+  }
+
+  /**
+   * @dev Returns the attestation IDs for a given subject and schema.
+   * @param subject The subject to retrieve attestation IDs for.
+   * @param schemaId The schema ID to retrieve attestation IDs for.
+   * @return An array of attestation IDs.
+   */
+  function getAttestationIdsBySubjectBySchema(
+    bytes memory subject,
+    bytes32 schemaId
+  ) public view returns (bytes32[] memory) {
+    return attestationIdsBySubjectBySchema[subject][schemaId];
+  }
+
+  /**
+   * @dev Returns the attestation IDs for a given attester.
+   * @param attester The attester to retrieve attestation IDs for.
+   * @return An array of attestation IDs.
+   */
+  function getAttestationIdsByAttester(address attester) public view returns (bytes32[] memory) {
+    return attestationIdsByAttester[attester];
+  }
+
+  /**
+   * @dev Returns the attestation IDs for a given schema.
+   * @param schema The schema to retrieve attestation IDs for.
+   * @return An array of attestation IDs.
+   */
+  function getAttestationIdsBySchema(bytes32 schema) public view returns (bytes32[] memory) {
+    return attestationIdsBySchema[schema];
+  }
+
+  /**
+   * @dev Returns the attestation IDs for a given portal.
+   * @param portal The portal to retrieve attestation IDs for.
+   * @return An array of attestation IDs.
+   */
+  function getAttestationIdsByPortal(address portal) public view returns (bytes32[] memory) {
+    return attestationIdsByPortal[portal];
+  }
+
+  /**
+   * @dev Returns the indexed status of an attestation.
+   * @param attestationId The ID of the attestation to check.
+   * @return The indexed status of the attestation.
+   */
+  function getIndexedAttestationStatus(bytes32 attestationId) public view returns (bool) {
+    return indexedAttestations[attestationId];
+  }
+
+  /**
+   * @dev Indexes an attestation.
+   * @param attestation The attestation to index.
+   */
+  function _indexAttestation(Attestation memory attestation) internal {
```

Comment:
is there a way to allow developers to specify which property they want to index by?  It would save on gas.  Of course, I guess they could just fork the module.